packages/mpt/src/proof/range.ts

```diff
@@ -412,17 +415,23 @@ async function hasRightElement(trie: MerklePatriciaTrie, key: Nibbles): Promise<
  * @param keys - key list of leaf data being proven.
  * @param values - value list of leaf data being proven, one-to-one correspondence with keys.
  * @param proof - proof node list, if all-elements-proof where no proof is needed, proof should be null, and both `firstKey` and `lastKey` must be null as well
+ * @param opts - optional, the opts may include a custom hashing function to use with the trie for proof verification
  * @returns a flag to indicate whether there exists more trie node in the trie
  */
-export async function verifyRangeProof(
+export async function verifyMerkleRangeProof(
   rootHash: Uint8Array,
-  firstKey: Nibbles | null,
-  lastKey: Nibbles | null,
-  keys: Nibbles[],
+  firstKeyRaw: Uint8Array | null,
+  lastKeyRaw: Uint8Array | null,
+  keysRaw: Uint8Array[],
   values: Uint8Array[],
   proof: Uint8Array[] | null,
-  useKeyHashingFunction: HashKeysFunction,
+  useKeyHashingFunction: HashKeysFunction = keccak256,
 ): Promise<boolean> {
+  // Convert Uint8Array keys to nibbles
+  const firstKey = firstKeyRaw !== null ? bytesToNibbles(firstKeyRaw) : null
```

Comment:
Doesn't the change from Nibbles to Uint8Array decrease the accuracy here? Nibbles are limited from the numbers 0 -0x0f and the uint8array is limited (per entry) from 0x0 to 0xff. (Trying to understand this change here and if it is limiting)

Reply:
Ah, wait, this method was never exported on package-level in the first place (it was only used internally) so this is fine.